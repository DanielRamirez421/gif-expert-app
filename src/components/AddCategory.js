import React, { Fragment, useState } from 'react'
import { PropTypes } from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputState, setInputState] = useState('')

    const handleInputChange = (e) => {
        setInputState(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        
        if ( inputState.trim().length > 2 ) {
            setCategories( cats => [ inputState, ...cats ] );
            setInputState( '' )
        }
    }

    return (
        <Fragment>
            <form onSubmit={ handleSubmit }>
                <input 
                    type="text"
                    value={ inputState }
                    onChange={ handleInputChange }/>
            </form>
        </Fragment>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
