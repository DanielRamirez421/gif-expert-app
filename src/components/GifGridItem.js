import React from 'react'

export const GifGridItem = ( {id, title, url} ) => {

    return (
        <div className='card'>
            <div className='card--image'>
                <img src={ url } alt={ title } />
            </div>
            <p><i>{ title }</i></p>
        </div>
    )
}
