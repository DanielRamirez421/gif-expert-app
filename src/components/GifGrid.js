import React, { Fragment } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGifs( category );

    return (
        <Fragment>
            <h3>{ category }</h3>

            <p className='not-results'>{ loading && 'Loading ...' }</p>
            <p className='not-results'>{ (!images.length && !loading) && (`Not results for: ${ category }`) }</p>

            <div className='card-grid'>
                {
                    images.map((img) => 
                        <GifGridItem
                            key={ img.id } 
                            { ...img }
                        />
                    )
                }
            </div>
        </Fragment>
    )
}
