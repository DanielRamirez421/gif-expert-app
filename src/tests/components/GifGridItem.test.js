import { GifGridItem } from "../../components/GifGridItem"
import { shallow } from 'enzyme';

describe('Testing component GridItem', () => {
    
    const title = 'a title';
    const url = 'http://localhost/algo.jpg';
    let wrapper = shallow( <GifGridItem title={ title } url={ url }/> );

    test('should show GifFridItem component correctly', () => {
        
        expect( wrapper ).toMatchSnapshot();

    });

    test('should have a paragraph with title', () => {
        
        const p = wrapper.find('p');
        const valueTitle = p.text().trim();
        expect( valueTitle ).toBe( title );

    });
    
    test('should have as src the url and alt the image title', () => {

        const image = wrapper.find('img');
        const imageSrc = image.prop('src');
        const imageAlt = image.prop('alt');

        expect(imageSrc).toBe(url);
        expect(imageAlt).toBe(title);
        
    });

    test('should have a class named animate__fadeInUp', () => {
       
        const div = wrapper.find('div').at(0);
        const desiredClass = 'animate__fadeInUp';
        const divClassNames = div.prop( 'className' );
        
        expect( div.hasClass( desiredClass ) ).toBe(true);
        expect( divClassNames ).toContain( desiredClass );

    });
        
    
})

