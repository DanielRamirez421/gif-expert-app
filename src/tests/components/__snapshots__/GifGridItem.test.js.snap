// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component GridItem should show GifFridItem component correctly 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <div
    className="card--image"
  >
    <img
      alt="a title"
      src="http://localhost/algo.jpg"
    />
  </div>
  <p>
    <i>
      a title
    </i>
  </p>
</div>
`;

exports[`Testing component GridItem should show GifGridItem component correctly 1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <div
    className="card--image"
  >
    <img
      alt="a title"
      src="http://localhost/algo.jpg"
    />
  </div>
  <p>
    <i>
      a title
    </i>
  </p>
</div>
`;
